### PR TITLE
Add info about new FormProtectionComponent.

### PR DESCRIPTION
**Pull request by ADmad**
Refs cakephp/cakephp#13728

### CODE REVIEW

File: en/controllers/components/form-protection.rst [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+    When using the FormProtection Component you **must** use the FormHelper to create
+    your forms. In addition, you must **not** override any of the fields' "name"
+    attributes. The FormProtection Component looks for certain indicators that are
+    created and managed by the FormHelper (especially those created in
+    :php:meth:`~Cake\\View\\Helper\\FormHelper::create()` and
+    :php:meth:`~Cake\\View\\Helper\\FormHelper::end()`).  Dynamically altering
+    the fields that are submitted in a POST request (e.g.  disabling, deleting
+    or creating new fields via JavaScript) is likely to cause the request to be
+    send to the blackhole callback.
```
Comment (markstory) [inline en/controllers/components/form-protection.rst]: It isn't called blackhole anymore is it?

File: en/controllers/components/form-protection.rst [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+unlockedActions
+    Actions to exclude from POST validation checks.
+
+validationFailureCallback
+    Callback to call in case of validation failure. Must be a valid Closure.
+    Unset by default in which case exception is thrown on validation failure.
+
+Disabling form tampering checks
+===============================
+
```
Comment (markstory) [inline en/controllers/components/form-protection.rst]:
```suggestion
::

```

Need this to make the next block a code block.

File: en/appendices/4-0-migration-guide.rst [hunk 3]
```diff
@@ -283,6 +284,8 @@ Helper
   text and HTML attributes by default. A new option ``escapeTitle`` has been
   added to allow controlling escaping the title separately from other HTML
   attributes.
+* ``Cake\View\Helper\SecureFieldTokenTrait`` has been removed. It's form token
```
Comment (garas) [inline en/appendices/4-0-migration-guide.rst]:
```suggestion
* ``Cake\View\Helper\SecureFieldTokenTrait`` has been removed. Its form token
```

File: en/controllers/components/form-protection.rst [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+works::
+
+    public function beforeFilter(EventInterface $event)
+    {
+        parent::beforeFilter($event);
+
+        $this->FormProtection->setConfig(
+            'validationFailureCallback',
+            function (BadRequestException $exception) {
+                // You are either return a response instance or throw the exception
```
Comment (garas) [inline en/controllers/components/form-protection.rst]:
```suggestion
                // You can either return a response instance or throw the exception
```

File: en/controllers/components/form-protection.rst [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+            $this->loadComponent('FormProtection');
+        }
+
+        public function beforeFilter(EventInterface $event)
+        {
+            parent::beforeFilter($event);
+
+            if ($this->request->getParam('prefix') === 'admin') {
+                $this->Security->setConfig('validate', false);
```
Comment (saeideng) [inline en/controllers/components/form-protection.rst]: why `->Security` ??

File: en/controllers/components/form-protection.rst [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+
+    use App\Controller\AppController;
+    use Cake\Event\EventInterface;
+
+    class WidgetController extends AppController
+    {
+        public function initialize(): void
+        {
+            parent::initialize();
+            $this->loadComponent('Security');
```
Comment (saeideng) [inline en/controllers/components/form-protection.rst]: `Security` component ? or `FormProtection`

File: en/controllers/components/form-protection.rst (hunk 5 above)
Comment (ADmad) [reply en/controllers/components/form-protection.rst]: Because i missed updating some references after copying from SecurityComponent page :)

**Pull request merged**